app/controllers/checkouts_controller.rb

```diff
@@ -0,0 +1,78 @@
+class CheckoutsController < ApplicationController
+  before_action :set_variables, only: [:checkout_current_card, :checkout_new_card]
+
+
+  def new
+    @product = Product.find(params[:id])
+    @cards = Card.where(user: current_user)
+  end
+
+
+  def checkout_current_card
+    @charge = Stripe::Charge.create(
+      :customer => params[:customer_id],
+      :amount => @amount,
+      :currency => 'eur'
+    )
+
+    record_purchase
+
+  rescue Stripe::CardError => e
+    flash[:error] = e.message
+    redirect_to root_path
+  end
+
+
+  def checkout_new_card
+    # create the customer account on stripe
+    customer = Stripe::Customer.create(
+      :email => params[:stripeEmail],
+      :source  => params[:stripeToken]
+    )
+
+    @charge = Stripe::Charge.create(
+      :customer    => customer.id,
+      :amount      => @amount,
+      :description => 'Marketplace customer',
+      :currency    => 'eur'
+    )
+
+    # store the user and the customer_id from stripe so we can charge the card automatically
+    card = Card.create(user: current_user, stripe_customer_id: customer.id, stripe_card_id: @charge[:source][:id], card_brand: @charge[:source][:brand], card_last_4: @charge[:source][:last4])
```

Comment:
@danielpaul Why does this make a difference, because I'm creating a new card here for that user?